rescript/_utilities.py

```diff
@@ -107,9 +107,19 @@ def _read_dna_fasta(path):
     return skbio.read(path, format='fasta', constructor=skbio.DNA)
 
 
+def _read_dna_alignment_fasta(path):
+    return skbio.TabularMSA.read(path, format='fasta', constructor=skbio.DNA)
+
+
 def _rna_to_dna(path):
     ff = DNAFASTAFormat()
     with ff.open() as outfasta:
         for seq in _read_rna_fasta(path):
             seq.reverse_transcribe().write(outfasta)
     return ff
+
+
+def _dna_iterator_to_aligned_fasta(iterator):
+    ff = AlignedDNAFASTAFormat()
+    skbio.io.write(iter(iterator), format='fasta', into=str(ff))
+    return ff
```

Comment:
Why is this not structured similar to `_rna_to_dna` above? or vice-versa? Is there something I am missing?

I am not sure if it is this step, or another, but this PR is now using more memory compared to  prior implementation. 🤷‍♂️   I am testing this out on the full NR99 SILVA alignment.

Reply:
Hmm, that's odd. I'll try to dig a bit more to see what the cause is.

About that piece of code: that's how this kind of conversion was done in q2-types so I borrowed it from there. Can try to re-write to resemble what you mentioned and see whether that had any impact.

Reply:
I have seen examples in the skbio cookbooks like this... I think the structure is fine, but @misialq if you do a quick benchmark it would be great to know 😄